go/src/meguca/db/util.go

```diff
@@ -85,6 +85,139 @@ func StartTransaction() (*sql.Tx, error) {
 	return db.Begin()
 }
 
+// WritePyu creates a new board's pyu row. Only used on board creation
+func WritePyu(b string) (err error) {
+	_, err = sq.Insert("pyu").
+		Columns("id", "pcount").
+		Values(b, 0).
+		Exec()
+
+	return
+}
+
+// GetPcount retrieves the board's pyu counter
+func GetPcount(b string) (c uint64, err error) {
+	err = sq.Select("pcount").
+		From("pyu").
+		Where("id = ?", b).
+		Scan(&c)
+
+	return
+}
+
+// GetPcount retrieves the board's pyu counter atomically
+func GetPcountA(tx *sql.Tx, b string) (c uint64, err error) {
+	r, err := withTransaction(tx, sq.Select("pcount").
+		From("pyu").
+		Where("id = ?", b)).
+		QueryRow()
+	r.Scan(&c)
```

Comment:
You are ignoring any scanning errors and .Scan() can be chained as:
```go
err =  withTransaction(tx, sq.Select("pcount").
		From("pyu").
		Where("id = ?", b)).
		QueryRow().
                Scan(&c)
```

Same for all cases bellow.

Reply:
I tried that, it caused a compile error since `withTransaction` returns two types.
That's why I had it scan the first variable.

Reply:
`go/src/meguca/db/util.go:113:11: multiple-value withTransaction(tx, sq.Select("pcount").From("pyu").Where("id = ?", b)).QueryRow() in single-value context`

Reply:
Also aren't the errors always handled outside of these functions?
If not, what's the point of the `if err != nil {` spam?

Reply:
Never mind I see what you mean now, I need to add the check now since it has it's own variable now and isn't checked for errors.